boards/metro_m0/examples/usb_serial.rs

```diff
@@ -181,75 +37,74 @@ fn main() -> ! {
     let mut red_led = pins.d13.into_open_drain_output(&mut pins.port);
     red_led.set_high().unwrap();
 
-    dbgprint!("make usb_bus");
+    //dbgprint!("make usb_bus");
 
-    let bus = unsafe {
-        USB_BUS = Some(hal::usb_bus(
-            peripherals.USB,
-            &mut clocks,
-            &mut peripherals.PM,
-            pins.usb_dm,
-            pins.usb_dp,
-            &mut pins.port,
+    let bus_allocator = unsafe {
+        USB_ALLOCATOR = Some(usb_allocator(
+                peripherals.USB,
+                &mut clocks,
+                &mut peripherals.PM,
+                pins.usb_dm,
+                pins.usb_dp,
+                &mut pins.port,
         ));
-        USB_BUS.as_ref().unwrap()
+        USB_ALLOCATOR.as_ref().unwrap()
     };
 
-    dbgprint!("make serial");
-    let serial = unsafe {
-        USB_SERIAL = Some(cdc_acm::SerialPort::new(bus));
-        USB_SERIAL.as_ref().unwrap()
+    unsafe {
+        USB_BUS = Some(
+            UsbDeviceBuilder::new(&bus_allocator, UsbVidPid(0x16c0, 0x27dd))
+                .manufacturer("Fake company")
+                .product("Serial port")
+                .serial_number("TEST")
+                .device_class(USB_CLASS_CDC)
+                .build(),
+            );
     };
 
-    dbgprint!("make dev");
+    //dbgprint!("make serial");
     unsafe {
-        USB_DEV = Some(
-            UsbDevice::new(bus, UsbVidPid(0x5824, 0x27dd))
-                .manufacturer("Wez Furlong")
-                .product("Serial port")
-                .serial_number("RUST")
-                .device_class(cdc_acm::USB_CLASS_CDC)
-                .build(&[serial]),
-        );
-    }
+        USB_SERIAL = Some(SerialPort::new(&bus_allocator));
+    };
 
     unsafe {
-        core.NVIC.set_priority(hal::Interrupt::USB, 0);
+        core.NVIC.set_priority(interrupt::USB, 0);
     }
-    core.NVIC.enable(hal::Interrupt::USB);
+    core.NVIC.enable(interrupt::USB);
 
-    dbgprint!("do loop");
+    //dbgprint!("do loop");
     loop {}
 }
 
 fn poll_usb() {
-    unsafe { USB_DEV.as_ref() }.map(|dev| {
-        dev.poll();
-
-        if dev.state() == UsbDeviceState::Configured {
-            let mut buf = [0u8; 8];
-            let serial = unsafe { USB_SERIAL.as_ref().unwrap() };
-
-            match serial.read(&mut buf) {
-                Ok(count) if count > 0 => {
-                    // Echo back in upper case
-                    for c in buf[0..count].iter_mut() {
-                        if 0x61 <= *c && *c <= 0x7a {
-                            *c &= !0x20;
+    unsafe {
+        USB_BUS.as_mut().map(|usb_dev| {
```

Comment:
What do you think about a `match` on USB_BUS and USB_SERIAL instead of two `map` calls to have less nesting?

Reply:
Sure, let me see what I can do with this one.

Reply:
I'm not sure if this is what you had in mind, but I changed the maps to a match in 8d3fedb.

Reply:
:+1: that’s what I had in mind 😄